### ISSUE TITLE
GsonRequestBodyConverter.java line 47 causing a lot of ANRs and 99% of them occurs on Android 12

### ISSUE THREAD

**Issue by EnesKy** [square/retrofit#3765]
What kind of issue is this?

**In our app, this issue has 286454 ANR events affecting 182342 users just in last 90 days.**
It's mostly happening in **Android 12 (%99).**
I couldn't find out why this causes ANR yet

We are using GsonConverter like this:
Retrofit
    .Builder().
    ...
    .addConverterFactory(GsonConverterFactory.create())

And here is our retrofit and othttp3 implemetations
    // Retrofit & Okhttp3
    implementation 'com.squareup.okhttp3:okhttp:4.9.3'
    implementation 'com.squareup.okhttp3:logging-interceptor:4.9.3'
    implementation('com.squareup.okhttp3:okhttp-urlconnection:4.9.1') {
        exclude group: 'com.squareup.okhttp3', module: 'okhttp'
    }
    implementation('com.squareup.retrofit2:retrofit:2.9.0') {
        exclude group: 'com.squareup.okhttp3', module: 'okhttp'
    }
    implementation 'com.squareup.retrofit2:converter-scalars:2.9.0'
    implementation 'com.squareup.retrofit2:converter-gson:2.9.0'
    implementation 'com.squareup.retrofit2:adapter-rxjava2:2.9.0'

_Can someone help me with this issue?_

Screenshots from Crashlytics Dashboard

<img width="1500" alt="Screen Shot 2022-08-23 at 11 14 38" src="https://user-images.githubusercontent.com/25669076/186107786-979a7116-1203-4595-a22a-3e30f28181e7.png">

<img width="1258" alt="Screen Shot 2022-08-23 at 11 15 17" src="https://user-images.githubusercontent.com/25669076/186107911-3fc792f2-3570-47b7-8d25-9d706c0f7110.png">
<img width="1286" alt="Screen Shot 2022-08-23 at 11 15 42" src="https://user-images.githubusercontent.com/25669076/186108006-8b89e085-d62f-4d3b-829c-fdfdc58d40ec.png">

**Comment by JakeWharton** [comment]
It's impossible to see without the full trace, but Retrofit is just calling into Gson here. It looks like some kind of crash reporting library is using Retrofit which is calling into Gson which is using reflection which is eventually calling into ICU and that's waiting on some lock.

Retrofit plays no part in the ANR here. Someone asked it to send a request and its asking Gson to serialize the request body object. You would see the same errors if you bypassed Retrofit and called OkHttp and Gson directly.

**Comment by EnesKy** [comment]
Gotcha 
I'll investigate deeper.
I'll add my notes here if any updates happen about this issue.

**Comment by EnesKy** [comment]
Hi @kmayoral,
Yes it did resolved for me.
We found that it was caused by a different library as @JakeWharton said.
I guess that library was creating a crash log that's too big to handle by GsonRequestBodyConverter which is working on main thread and it was causing ANR.
We talked to the team that developed the library and they solved the problem on their side after that i just updated the library with the new version.
Sorry for late response :)
